src/Chorus/UI/Clone/GetCloneFromInternetDialog.cs

```diff
@@ -261,11 +261,27 @@ private void _cancelButton_Click(object sender, EventArgs e)
 		}
 
 		private void OnDownloadClick(object sender, EventArgs e)
+		{
+			StartClone();
+		}
+
+		public void StartClone(string username, string password, Uri targetUri)
+		{
+			_model.Username = username;
+			_model.Password = password;
+			_model.CustomUrl = targetUri.ToString();
+			_model.IsCustomUrl = true;
+			_model.LocalFolderName = GetProjectName(targetUri);
```

Comment:
I'd rather have this passed in explicitly then trying to determine it from the uri, if the uri format changes then this will need to be rewritten. Right now it depends on the project name being somewhere specific in the uri but there's not really anything else that requires that to be the case.

Reply:
Sure, done.